src/meta/src/barrier/mod.rs

```diff
@@ -684,14 +694,21 @@ impl GlobalBarrierManager {
                         _ => {}
                     }
                 }
-                resp_result = self.control_stream_manager.next_complete_barrier_response() => {
+                (worker_id, resp_result) = self.control_stream_manager.next_complete_barrier_response() => {
                     match resp_result {
-                        Ok((worker_id, prev_epoch, resp)) => {
-                            self.checkpoint_control.barrier_collected(worker_id, prev_epoch, resp);
+                        Ok(resp) => {
+                            self.checkpoint_control.barrier_collected(resp);
 
                         }
                         Err(e) => {
-                            self.failure_recovery(e).await;
+                            if self.checkpoint_control.on_collect_failure(worker_id, &e)
+                                || self.state.inflight_actor_infos.actor_map.contains_key(&worker_id) {
```

Comment:
I don't think we need to return a bool for `self.checkpoint_control.on_collect_failure`. Simply checking `self.state.inflight_actor_infos.actor_map` seems to be sufficient if the purpose here is to detect whether there is any actor allocated for a worker.

Reply:
The `self.state.inflight_actor_infos.actor_map` stores the latest map that has been applied with mutation all the injected barrier. Therefore, for example, if we have a drop mview command that drops all mviews, after its barrier is injected and wait for collection, the `self.state.inflight_actor_infos.actor_map` is empty, and only checking it will ignore this error, but we should not ignore it.